ui/dialogs/settingsdialog.go

```diff
@@ -296,6 +309,10 @@ func (s *SettingsDialog) createGeneralTab(canSaveQueueToServer bool) *container.
 		saveQueueHBox,
 		trackNotif,
 		albumGridYears,
+		container.NewHBox(
+			widget.NewLabel(lang.L("Queue size: ")),
```

Comment:
I think this needs a more descriptive settings name. "Queue size" is kind-of ambiguous, maybe something like "Random queue size"?

Reply:
Done

Reply:
Although now that I remember, that setting also applies for "play similar artists". Should be better to remove it?

Reply:
We could remove it from the settings dialog but leave it in the config file. There are already a few settings that are config-file-only (which most people probably won't need to change)

Reply:
I think some users could find it useful to easily change it, but maybe it's just my personal preference. I remove it from the UI.

btw, there is plans to improve the performance regarding the lists?

Reply:
I think there is room for improvement on loading the saved play queue at startup, but I am not sure how much improvement there can be on playing random songs, since most of the time is waiting for and parsing the server response

Reply:
The bottleneck seems to be more the parsing that waiting for the reply. In the next graph it shows that ~1/3 is making the HTTP request and 2/3 is parsing the XML. 
![imatge](https://github.com/user-attachments/assets/3353ed8d-7abd-42dc-b9ee-974e5fd45823)

Reply:
Well aside from switching away from Go's standard XML parser, or reworking the whole subsonic library to use JSON (including the work to handle date idiosyncracies across different servers), there's not much to be done. Switching to JSON would be something I would review if someone else were to do the work but it will be low on my personal priority list.

Reply:
To add onto this, it would not be a very large amount of work to switch to JSON, but the custom marshalXML/unmarshalXML implementations for the xsdDateTime type in [here](https://github.com/supersonic-app/go-subsonic/blob/master/subsonic/models.go) would need to be implemented and tested with the JSON parser as well. All the workarounds for things like timestamps with and without timezone suffixes would have to continue to work

Reply:
Do you think it would be feasible and easier to use concurrent jobs?

Reply:
Queuing random/similar tracks from the server is a single HTTP call so I don't think it's really parallelizable. On the other hand, loading the saved play queue currently issues one HTTP call per track, to get updated play count and other metadata, so concurrent calls could likely speed that up significantly

Reply:
I see. Thank you. 

btw I changed the config name as requested